### PR TITLE
feat: override get_head search request metadata

### LINKED ISSUE #537: Unable to fetch head metadata on search strategy when search keys has more than one word
### Describe the bug

There seem to be a few issues with the Yoast get_head endpoint that is not properly recognizing the requested page as a search page.

### Steps to Reproduce

1. Go To Search
2. Search for two words
3. Missing/wrong metatags or crash.

### Screenshots, screen recording, code snippet

_No response_

### Environment information

_No response_

### WordPress information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by lucymtc**
### Description of the Change
Yoast does not yoast_head to the WordPress search endpoint.
Yoast is not properly detecting a search page via the get_head function.

These changes fix the SEO title and canonical URL for the search results when requesting to yoast get_head endpoint.
Note: I have installed a fresh instance of WP to test the get_head endpoint for search results and the response matches the home yoast_head. However running WP in wp-env inside of headstartwp the response is 404 for unknown reason.

Closes https://github.com/10up/headstartwp/issues/537

### Changelog Entry

> Added - New feature
> Changed - Existing functionality
> Deprecated - Soon-to-be removed feature
> Removed - Feature
> Fixed - Bug fix
> Security - Vulnerability

### Credits

Props @username, @username2, ...

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: wp/headless-wp/includes/classes/Integrations/YoastSEO.php [hunk 1]
```diff
@@ -162,4 +168,114 @@ public function maybe_override_sitemap_robots_url( $output ) {
...
+	 *
+	 * @return boolean|array False if it's not a yoast search rest api request. Search $query_vars if otherwise.
+	 */
+	public function get_yoast_search_query_vars() {
+
+		if ( ! ( defined( 'REST_REQUEST' ) && REST_REQUEST ) ) {
+			return false;
+		}
+
+		$request_uri = isset( $_SERVER['REQUEST_URI'] ) ? sanitize_url( wp_unslash( $_SERVER['REQUEST_URI'] ) ) : '';
```
Comment (nicholasio) [inline wp/headless-wp/includes/classes/Integrations/YoastSEO.php]: sanitize_url is deprecated. Looks like `esc_url` is the alternative.

File: wp/headless-wp/includes/classes/Integrations/YoastSEO.php [hunk 2]
```diff
@@ -162,4 +168,114 @@ public function maybe_override_sitemap_robots_url( $output ) {
...
+	 * @return string
+	 */
+	public function replace_yoast_search_title_placeholders( $title, $query_vars ) {
+
+		$str_replace_mapping = apply_filters(
+			'tenup_headless_wp_search_title_variables_replacments',
+			array(
+				'%%sitename%%'     => get_bloginfo( 'name' ),
+				'%%searchphrase%%' => $query_vars['s'] ?? '',
+				' %%page%%'        => '',
```
Comment (nicholasio) [inline wp/headless-wp/includes/classes/Integrations/YoastSEO.php]: Do you know what we need to support pages? Looks like it's not being parsed from query vars.

**Review by nicholasio — COMMENTED** [review]
Tested and it works fine for single words searches.

It would be great to support pagination.

File: wp/headless-wp/includes/classes/Integrations/YoastSEO.php (hunk 2 above)
Comment (lucymtc) [reply wp/headless-wp/includes/classes/Integrations/YoastSEO.php]: not quite yet, I left a todo https://github.com/10up/headstartwp/pull/663/files#diff-39367dca2db0a8e747c2637f520848d9dd3b1321023b9eff03a75c0c51a524a7R208 comment, I can take a look next. Probably we need to get the page number with regex

Comment (lucymtc) [reply wp/headless-wp/includes/classes/Integrations/YoastSEO.php]: @nicholasio I have pushed a change to support page number. 
However there is a limitation, in a regular WP the search title will show as "Page 2 of 10 ", we don't have the information to know how many pages there are. So it only shows "Page 2"

File: wp/headless-wp/includes/classes/Integrations/YoastSEO.php (hunk 1 above)
Comment (lucymtc) [reply wp/headless-wp/includes/classes/Integrations/YoastSEO.php]: changed

**Comment by nicholasio** [comment]
@lucymtc  Can you add a changeset?

**Review by nicholasio — APPROVED** [review]

**Pull request merged**